Client/Frontend/Browser/ClipboardBarDisplayHandler.swift

```diff
@@ -7,17 +7,22 @@ public struct ClipboardBarToastUX {
 }
 
 protocol ClipboardBarDisplayHandlerDelegate: class {
-    func shouldDisplayClipboardBar(absoluteString: String)
+    func shouldDisplayClipboardBar(_ clipboardBar: ButtonToast)
 }
 
 class ClipboardBarDisplayHandler {
     weak var delegate: ClipboardBarDisplayHandlerDelegate?
+    weak var settingsDelegate: SettingsDelegate?
```

Comment:
both delegate and settingDelegate are the same object. I think you can do 
`var delegate: (ClipboardBarDisplayHandlerDelegate & SettingsDelegate)?` I'm not 100% sure but something like that should work.